dev/com.ibm.websphere.appserver.features/test/src/com/ibm/ws/feature/tests/VisibilityTest.java

```diff
@@ -283,4 +288,69 @@ public void testVisibility2() {
         }
     }
 
+    /**
+     * Tests that an auto feature has more than one feature in its filter.
+     * Tests that an auto feature doesn't depend on another auto feature.
+     */
+    @Test
+    public void testAutoFeatures() {
+        StringBuilder errorMessage = new StringBuilder();
+        for (Entry<String, FeatureInfo> entry : features.entrySet()) {
+            String featureName = entry.getKey();
+            FeatureInfo featureInfo = entry.getValue();
+
+            if (!featureInfo.isAutoFeature()) {
+                continue;
+            }
+
+            String[] filterFeatures = featureInfo.getAutoFeatures();
+            if (filterFeatures.length == 1) {
+                errorMessage.append("Found issues with " + featureName + '\n');
+                errorMessage.append("     Auto feature filter only depends on one feature " + filterFeatures[0] + ".\n");
+                errorMessage.append("     The feature and/or bundle dependencies in this auto feature should just be a dependency of that feature\n");
+                errorMessage.append("     OR this should be turned into a private feature that " + filterFeatures[0] + " depends on.");
+            }
+        }
+        if (errorMessage.length() != 0) {
+            Assert.fail("Found auto features that have edition set incorrectly: " + '\n' + errorMessage.toString());
```

Comment:
Is this a copy paste error?  This checking doesn't seem to have anything to do with `edition`.  should it say `that have the filter set incorrectly`?